packages/react-menu/src/components/Menu/useMenu.tsx

```diff
@@ -59,6 +72,86 @@ export const useMenu = (props: MenuProps, ref: React.Ref<HTMLElement>, defaultPr
     },
     [setOpen],
   );
+  state.on.length === 0 && state.on.push('click');
+
+  useMenuPopup(state);
+  useOnClickOutside({ element: document, refs: [state.menuPopupRef, triggerRef], callback: () => setOpen(false) });
+  useOnEscape({ element: document, callback: () => setOpen(false) });
+  return state;
+};
+
+const useMenuPopup = (state: MenuState) => {
+  // TODO use Popper for the popup ref
+  const menuPopupRef = React.useRef<HTMLElement>(null) as React.MutableRefObject<HTMLElement>;
+  state.menuPopupRef = menuPopupRef;
+  state.menuPopup.children = (Component, originalProps) => {
+    const newProps = { 'aria-labelledBy': state.triggerId, ...originalProps };
+
+    if (state.on.includes('hover')) {
+      newProps.onMouseEnter = (e: React.MouseEvent<HTMLElement>) => {
+        state.setOpen(true);
+        originalProps?.onMouseEnter?.(e);
+      };
+      newProps.onMouseLeave = (e: React.MouseEvent<HTMLElement>) => {
+        state.setOpen(false);
+        originalProps?.onMouseLeave?.(e);
+      };
+    }
+
+    newProps.onBlur = (e: React.FocusEvent<HTMLElement>) => {
+      state.setOpen(false);
+      originalProps?.onBlur?.(e);
+    };
+
+    return (
+      // Shorthand render funct types have issues
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      <Component {...newProps} ref={state.menuPopupRef}>
+        {state.menuList}
+      </Component>
+    );
+  };
 
   return state;
 };
+
+// TODO handle nested menus
+const useOnClickOutside = (options: {
+  element?: Node | Window | Document;
+  refs: React.MutableRefObject<HTMLElement>[];
+  callback: (ev: MouseEvent) => void;
+}) => {
+  const { refs, callback, element = document } = options;
+
+  React.useEffect(() => {
+    const listener = (ev: MouseEvent) => {
+      const isOutside = !refs.some(ref => elementContains(ref.current, ev.target as HTMLElement));
+      if (isOutside) {
+        callback(ev);
+      }
+    };
+
+    element?.addEventListener('click', listener);
+
+    return () => element?.removeEventListener('click', listener);
+  }, [refs, callback, element]);
+};
+
+// TODO handle nested menus
+const useOnEscape = (options: { element?: Node | Window | Document; callback: (ev: KeyboardEvent) => void }) => {
+  const { callback, element = document } = options;
+
+  React.useEffect(() => {
+    const listener = (ev: KeyboardEvent) => {
+      const keyCode = getCode(ev);
+      if (keyCode === keyboardKey.Escape) {
+        callback(ev);
+      }
+    };
+
+    element?.addEventListener('keydown', listener);
+
+    return () => element?.removeEventListener('keydown', listener);
+  }, [callback, element]);
```

Comment:
I don't think that it is a desired behavior: as focus should be inside a `Menu`, it's fine to use a handler on it. And it's what Northstar does: 

https://github.com/microsoft/fluentui/blob/1592b8fa62e4bf34777ac2af7a9e0c7937d8b8b3/packages/fluentui/accessibility/src/behaviors/Menu/menuItemBehavior.ts#L59-L61